lua-lsp/methods.lua

```diff
@@ -191,6 +196,13 @@ local function make_completion_items(k, val, isField, isInvoke, isVariant)
 				table.insert(sig, " ??? ")
 			end
 
+			-- Construct a snippet insertion format from signature
```

Comment:
are there any online references to the snippet format we could include here? just so it's clear what the final string we are building is intended to look like

Reply:
I meant to get this in pre-merge but lost track of it. The snippet format is available here: https://microsoft.github.io/language-server-protocol/specification.html#snippet_syntax

And the format is described like so in the structure definition:

> A snippet can define tab stops and placeholders with `$1`, `$2`
> and `${3:foo}`. `$0` defines the final tab stop, it defaults to
> the end of the snippet. Placeholders with equal identifiers are linked,
> that is typing in one will update others too.